### DISCUSSION TITLE
How do I avoid rebuilding the table structure every time I execute dbt run?

### DISCUSSION THREAD

**Discussion by xy-qimu** [dbt-labs/dbt-core#9142] [Q&A]
materialized is table and how to avoid rebuiding the table structure every time I execute dbt run?

**Comment by edwrand** [comment]
You could materialize as incremental and it will only add or update rows that have changed since last run

`{{ config(materialized='incremental') }}`

**Reply by xy-qimu** [reply]
Thank you for your response.

However, I still have some doubts. Could you please help clarify:

If the goal is to avoid restructuring the table, using an incremental model might be suitable. However, this might not apply to all scenarios and could potentially lead to performance issues with larger datasets?

I'm more inclined to manually create table structures and then use dbt for data processing and loading. Would this approach not be considered best practice? Can I continue doing this without having to reconstruct the table structure every time I execute the process?
